server/src/main/java/com/linecorp/centraldogma/server/internal/storage/ScheduledPurgeService.java

```diff
@@ -36,64 +39,131 @@
 import com.google.common.util.concurrent.ListenableScheduledFuture;
 import com.google.common.util.concurrent.ListeningScheduledExecutorService;
 import com.google.common.util.concurrent.MoreExecutors;
+import com.spotify.futures.CompletableFutures;
 
 import com.linecorp.centraldogma.common.Author;
 import com.linecorp.centraldogma.server.command.Command;
 import com.linecorp.centraldogma.server.command.CommandExecutor;
+import com.linecorp.centraldogma.server.metadata.MetadataService;
+import com.linecorp.centraldogma.server.metadata.UserAndTimestamp;
 import com.linecorp.centraldogma.server.storage.project.ProjectManager;
 
 import io.netty.util.concurrent.DefaultThreadFactory;
 
 /**
  * A service class for pursing projects and repositories that were removed before.
  */
-public class StoragePurgeService {
+public class ScheduledPurgeService {
 
-    private static final Logger logger = LoggerFactory.getLogger(StoragePurgeService.class);
+    private static final Logger logger = LoggerFactory.getLogger(ScheduledPurgeService.class);
 
     /**
      * How often to run the storage purge schedules. i.e. every minute.
      */
-    private static final Duration TICK = Duration.ofSeconds(1);
+    private static final Duration TICK = Duration.ofMinutes(1);
+    private static final Duration INITIAL_DELAY = Duration.ofSeconds(1);
 
     private final ProjectManager projectManager;
     private final long maxRemovedRepositoryAgeMillis;
     private final StoragePurgeScheduler storagePurgeScheduler = new StoragePurgeScheduler();
 
-    public StoragePurgeService(
+    public ScheduledPurgeService(
             ProjectManager projectManager,
             long maxRemovedRepositoryAgeMillis) {
         this.projectManager = requireNonNull(projectManager, "projectManager");
-        checkArgument(maxRemovedRepositoryAgeMillis > 0,
-                      "maxRemovedRepositoryAgeMillis: %s (expected: > 0)", maxRemovedRepositoryAgeMillis);
+        checkArgument(maxRemovedRepositoryAgeMillis >= 0,
+                      "maxRemovedRepositoryAgeMillis: %s (expected: >= 0)", maxRemovedRepositoryAgeMillis);
         this.maxRemovedRepositoryAgeMillis = maxRemovedRepositoryAgeMillis;
     }
 
-    public void start(CommandExecutor commandExecutor) {
-       storagePurgeScheduler.start(() -> schedulePurgeData(commandExecutor));
+    public void start(CommandExecutor commandExecutor,
+                      MetadataService metadataService) {
+        if (!isDisabled()) {
+            requireNonNull(commandExecutor, "commandExecutor");
+            requireNonNull(metadataService, "metadataService");
+            cleanPurgedFiles(commandExecutor);
+            storagePurgeScheduler.start(() -> schedulePurgeData(commandExecutor, metadataService).join());
```

Comment:
- Could we start the scheduler task after the commands issued by `cleanPurgedFiles()` are all done? It's because otherwise two threads may try to issue `Purge*Command`s at the same time.
- Catch and log an exception.